projects/plugins/jetpack/extensions/blocks/ai-assistant/hooks/use-suggestions-from-openai/index.js

```diff
@@ -232,7 +240,57 @@ const useSuggestionsFromOpenAI = ( {
 			setWasCompletionJustRequested( false );
 		}
 
-		source?.current?.addEventListener( 'done', e => {
+		const onFunctionDone = async e => {
+			const { detail } = e;
+
+			// Add assistant message with the function call request
+			const assistantResponse = { role: 'assistant', content: null, function_call: detail };
+
+			const response = await implementedFunctions[ detail.name ]?.(
+				JSON.parse( detail.arguments )
+			);
+
+			// Add the function call response
+			const functionResponse = {
+				role: 'function',
+				name: detail?.name,
+				content: JSON.stringify( response ),
+			};
+
+			prompt = [ ...prompt, assistantResponse, functionResponse ];
+
+			// Remove source.current listeners
+			source?.current?.removeEventListener( 'functionDone', onFunctionDone );
+			source?.current?.removeEventListener( 'done', onDone );
+			source?.current?.removeEventListener( 'error_unclear_prompt', onErrorUnclearPrompt );
+			source?.current?.removeEventListener( 'error_network', onErrorNetwork );
+			source?.current?.removeEventListener(
+				'error_service_unavailable',
+				onErrorServiceUnavailable
+			);
+			source?.current?.removeEventListener( 'error_quota_exceeded', onErrorQuotaExceeded );
+			source?.current?.removeEventListener( 'error_moderation', onErrorModeration );
+			source?.current?.removeEventListener( 'suggestion', onSuggestion );
+
+			source.current = await askQuestion( prompt, {
+				postId,
+				requireUpgrade,
+				feature: attributes?.useGpt4 ? 'ai-assistant-experimental' : null,
+				functions: options.functions,
+			} );
+
+			// Add the listeners back
+			source?.current?.addEventListener( 'functionDone', onFunctionDone );
```

Comment:
The event-name shape isn't consistent with the rest of the events.
```suggestion
			source?.current?.addEventListener( 'function_done', onFunctionDone );
```